infrahub_sdk/client.py

```diff
@@ -143,13 +144,38 @@ def __init__(
 
         self.update_group_context = self.config.update_group_context
         self.identifier = self.config.identifier
+        self._cached_ssl_context: SSLContext | None = None
+        self._cached_ssl_context_source: str | None = None
         self.group_context: InfrahubGroupContext | InfrahubGroupContextSync
         self._initialize()
         self._request_context: RequestContext | None = None
 
     def _initialize(self) -> None:
         """Sets the properties for each version of the client"""
 
+    def _get_verify_config(self) -> bool | SSLContext:
```

Comment:
I'd suggest to move this to the config class instead and have it as a property there.

Then have something like:

```python
        async with httpx.AsyncClient(
            **proxy_config,  # type: ignore[arg-type]
            verify=self.config.tls_context,
        ) as client:
```